util/merkle/src/triedbmut.rs

```diff
@@ -0,0 +1,539 @@
+
+                                        match n_leaf_node {
+                                            None => Err(Box::new(TrieError::IncompleteDatabase(hash))),
+                                            Some(RlpNode::Leaf(child_partial, child_v)) => {
+                                                let n_partial =
+                                                    NibbleSlice::new_composed(&partial, index as u8, &child_partial);
+                                                let n_leaf = RlpNode::Leaf(n_partial, child_v);
+                                                let mut node_rlp = RlpNode::encoded(Some(n_leaf), None);
+                                                let n_hash = self.db.insert(&node_rlp[..]);
+
+                                                Ok(Some(n_hash))
+                                            }
+                                            Some(RlpNode::Branch(child_partial, child_key)) => {
+                                                let n_partial =
+                                                    NibbleSlice::new_composed(&partial, index as u8, &child_partial);
+                                                let n_branch = RlpNode::Branch(n_partial, child_key);
+                                                let mut node_rlp = RlpNode::encoded(Some(n_branch), None);
+                                                let n_hash = self.db.insert(&node_rlp[..]);
+
+                                                Ok(Some(n_hash))
+                                            }
+                                        }
+                                    }
+                                    _ => {
+                                        let n_branch = RlpNode::Branch(partial, key);
+                                        let mut node_rlp = RlpNode::encoded(Some(n_branch), None);
+                                        let n_hash = self.db.insert(&node_rlp[..]);
+
+                                        Ok(Some(n_hash))
+                                    }
+                                }
+                            } else {
+                                let n_branch = RlpNode::Branch(partial, key);
+                                let mut node_rlp = RlpNode::encoded(Some(n_branch), None);
+                                let n_hash = self.db.insert(&node_rlp[..]);
+
+                                Ok(Some(n_hash))
+                            }
+                        } else {
+                            Ok(cur_node_hash)
+                        }
+                    }
+                    None => Ok(cur_node_hash),
+                }
+            }
+            None => Ok(cur_node_hash),
+        }
+    }
+
+    /// Print Trie in format
+    pub fn fmt(&self, cur_node_hash: H256) {
+        let node_rlp = self.db.get(&cur_node_hash).expect("Broken Database");
+
+        let node = RlpNode::decoded(&node_rlp[..]);
+        println!("{:?}", node);
```

Comment:
Use [Formatter](https://doc.rust-lang.org/std/fmt/struct.Formatter.html) instead of `println`.

Reply:
I can not understand what you mean. So I just make 
```

impl<'a> fmt::Display for RlpNode<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            RlpNode::Leaf(partial, value) => writeln!(f, "Leaf - key({:?}), value({:?})", partial, value),
            RlpNode::Branch(partial, children) => {
                writeln!(f, "Branch - path({:?})", partial)?;

                for i in 0..16 {
                    writeln!(f, "child {} - hash({:?})", i, children[i])?;
                }
                Ok(())
            }
        }
    }
}
```
and print node with `println!`